### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

**Comment by vncoelho** [comment]
Looks good, Erik! Nice refactoring and at the right time.

File: neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -537,6 +520,16 @@ private void SendPrepareRequest()
             }
             ChangeTimer(TimeSpan.FromSeconds(Blockchain.SecondsPerBlock << (context.ViewNumber + 1)));
         }
+
+        private bool VerifyRequest()
+        {
+            if (!Blockchain.GetConsensusAddress(context.Snapshot.GetValidators(context.Transactions.Values).ToArray()).Equals(context.NextConsensus))
+                return false;
+            Transaction minerTx = context.Transactions.Values.FirstOrDefault(p => p.Type == TransactionType.MinerTransaction);
```
Comment (shargon) [inline neo/Consensus/ConsensusService.cs]: We should ensure that is the first TX

File: neo/Consensus/Helper.cs [hunk 2]
```diff
@@ -11,6 +12,23 @@ internal static class Helper
         /// </summary>
         public const byte CN_Context = 0xf4;
 
+        public static int F(this IConsensusContext context) => (context.Validators.Length - 1) / 3;
```
Comment (shargon) [inline neo/Consensus/Helper.cs]: We can add the `inline`  attribute (to all of them)

File: neo/Consensus/ConsensusService.cs (hunk 1 above)
Comment (erikzhang) [reply neo/Consensus/ConsensusService.cs]: https://github.com/neo-project/neo/blob/eaeb0a5b01f9a5c37d533125f21081f094f15975/neo/Consensus/PrepareRequest.cs#L39-L40

It is checked here.

File: neo/Consensus/Helper.cs (hunk 2 above)
Comment (erikzhang) [reply neo/Consensus/Helper.cs]: Added them in #606, please review it.

**Pull request merged**